BDArmory/Modules/ModuleWeapon.cs

```diff
@@ -3200,6 +3174,9 @@ public void RunTrajectorySimulation()
                                 if (simTime > thrustTime)
                                 {
                                     simDeltaTime = 0.1f;
+                                    //var timeOfFlight = BallisticTrajectorySimulation(ref simCurrPos, simVelocity, Mathf.Min(maxTargetingRange, targetDistance), ((simStartPos - simCurrPos).magnitude - maxTargetingRange) / simVelocity.magnitude, simDeltaTime, FlightGlobals.getAltitudeAtPos(targetPosition) < 0);
```

Comment:
Any particular reason why this is commented out?
I think using
```csharp
var currentTargetDistance = Mathf.Min(maxTargetingRange, (simCurrPos - targetPosition).magnitude);
simDeltaTime = Mathf.Clamp(currentTargetDistance / simVelocity.magnitude / 2f, Time.fixedDeltaTime, Time.fixedDeltaTime * BDArmorySettings.BALLISTIC_TRAJECTORY_SIMULATION_MULTIPLIER);
var timeOfFlight = BallisticTrajectorySimulation(ref simCurrPos, simVelocity, currentTargetDistance, maxTargetingRange / bulletVelocity - simTime, simDeltaTime, FlightGlobals.getAltitudeAtPos(targetPosition) < 0) + simTime;
bulletPrediction = simCurrPos;
break;
```
should give the correct result.

Reply:
It was commented out since it was something I was working on, but hadn't yet thoroughly tested. There's been a couple of instances where my testing was sloppier than it should have been, and missed things that needed to be fixed later, so this time I wanted to be cautious. But yes, in theory, it should work and give the correct result.